### ISSUE TITLE
Performance Analyzer fixes needed

### ISSUE THREAD

**Issue by inferno-chromium** [google/oss-fuzz#389]
1) Distinguish between types of coverage failures. E.g. atleast distinguish https://github.com/google/oss-fuzz/issues/382 into its own issue class (like instrumentation failure or something). Also need to see to distinguish between if current code is sufficient and not covering new pieces OR current code is insufficient and we need to write different fuzzers to cover functions.

2) https://bugs.chromium.org/p/chromium/issues/detail?id=692256
We should show both startup crash, logging issue
e.g. look at logs from https://pantheon.corp.google.com/storage/browser/gnutls-logs.clusterfuzz-external.appspot.com/libFuzzer_gnutls_idna_parser_fuzzer/libfuzzer_asan_gnutls/2017-02-12/

https://oss-fuzz.com/v2/performance-report/libFuzzer_gnutls_idna_parser_fuzzer/libfuzzer_asan_gnutls/latest shows MemoryError.. we might dump too much log.

**Comment by inferno-chromium** [comment]
Also see 
https://github.com/google/oss-fuzz/issues/382#issuecomment-279859178

If number of coverage points drops significantly in one of the logs and continue, we need alerts or whatever.

**Comment by Dor1s** [comment]
I'm trying to resolve point 2 and it iss tricky. If we got a few libFuzzer lines (< threshold) and a bunch of lines after them, we cannot understand if those lines are logging or some failure. We can rely on crash detection, but would be better to see more examples.

**Comment by inferno-chromium** [comment]
@tanin47 - i think we should definitely show MemoryError as logging issue (since i dont see another reason for it, maybe add FIXME to improve more). and any tracebacks should be ignored at performance analyzer level and instrad logged from fuzz_task level on our logging/status pipeline.

**Comment by Dor1s** [comment]
I've seen the following error when I've built fuzz target incorrectly:

```
# /out/convert_woff2ttf_fuzzer ./c1/
WARNING: Failed to find function "__sanitizer_print_stack_trace".
INFO: Seed: 1975231563
INFO: Loaded 0 modules (0 guards): 
Loading corpus dir: ./c1/
INFO: -max_len is not provided, using 64
INFO: A corpus is not provided, starting from an empty corpus
#0	READ units: 1
#1	INITED exec/s: 0 rss: 6Mb
ERROR: no interesting inputs were found. Is the code instrumented for coverage? Exiting.
```

Looks like we can rely on that message for a specific startup crash type.

**Comment by Dor1s** [comment]
We can detect startup crashes now, though we do not report them anywhere except of performance analysis page. We probably can raise an alert for startup crashes as they shouldn't happen often.

For the case https://github.com/google/oss-fuzz/issues/389#issuecomment-282495922, I consider adding one more performance flag based on `Is the code instrumented` substring. That would also address point 1 from the initial bug description.

**Comment by Dor1s** [comment]
As we are moving away from LogsAnalysis task, we cannot catch unhandled exceptions happening in the launcher (such as MemoryError or any other), but, we can do that before or during log uploading. Also we don't expect that to ever happen, so we definitely want to report such cases. I've filed a Notifications bug (https://github.com/google/oss-fuzz/issues/695).

Bad instrumentation detection has been added to existing performance analysis pipeline.

Tricky cases when implementation is partially broken should be detected in post build scripts, tracked by issue https://github.com/google/oss-fuzz/issues/382
